cardano_node_tests/tests/test_plutus_spend_build.py

```diff
@@ -653,6 +651,207 @@ def test_guessing_game(
+            redeemer_cbor_file=plutus_common.REDEEMER_42_TYPED_CBOR,
+        )
+
+        # Step 1: fund the Plutus scripts
+
+        script_address1 = cluster.gen_payment_addr(
+            addr_name=f"{temp_template}_addr1", payment_script_file=plutus_op1.script_file
+        )
+        datum_hash1 = cluster.get_hash_script_data(script_data_file=plutus_op1.datum_file)
+
+        script_address2 = cluster.gen_payment_addr(
+            addr_name=f"{temp_template}_addr2", payment_script_file=plutus_op2.script_file
+        )
+        datum_hash2 = cluster.get_hash_script_data(script_data_file=plutus_op2.datum_file)
+
+        # create a Tx output with a datum hash at the script address
+
+        tx_files_fund = clusterlib.TxFiles(
+            signing_key_files=[payment_addrs[0].skey_file],
+        )
+        txouts_fund = [
+            clusterlib.TxOut(
+                address=script_address1,
+                amount=script_fund,
+                datum_hash=datum_hash1,
+            ),
+            clusterlib.TxOut(
+                address=script_address2,
+                amount=script_fund,
+                datum_hash=datum_hash2,
+            ),
+            # for collateral
+            clusterlib.TxOut(address=payment_addrs[1].address, amount=collateral_fund),
+            clusterlib.TxOut(address=payment_addrs[1].address, amount=collateral_fund),
+        ]
+        tx_output_fund = cluster.build_tx(
+            src_address=payment_addrs[0].address,
+            tx_name=f"{temp_template}_step1",
+            tx_files=tx_files_fund,
+            txouts=txouts_fund,
+            fee_buffer=2_000_000,
+            join_txouts=False,
+        )
+        tx_signed_fund = cluster.sign_tx(
+            tx_body_file=tx_output_fund.out_file,
+            signing_key_files=tx_files_fund.signing_key_files,
+            tx_name=f"{temp_template}_step1",
+        )
+
+        cluster.submit_tx(tx_file=tx_signed_fund, txins=tx_output_fund.txins)
+
+        txid_fund = cluster.get_txid(tx_body_file=tx_output_fund.out_file)
+        script_utxos1 = cluster.get_utxo(txin=f"{txid_fund}#1", coins=[clusterlib.DEFAULT_COIN])
+        script_utxos2 = cluster.get_utxo(txin=f"{txid_fund}#2", coins=[clusterlib.DEFAULT_COIN])
+        collateral_utxos1 = cluster.get_utxo(txin=f"{txid_fund}#3")
+        collateral_utxos2 = cluster.get_utxo(txin=f"{txid_fund}#4")
+
+        assert script_utxos1 and script_utxos2, "No script UTxOs"
+        assert collateral_utxos1 and collateral_utxos2, "No collateral UTxOs"
+
+        assert (
+            script_utxos1[0].amount == script_fund
+        ), f"Incorrect balance for script address `{script_utxos1[0].address}`"
+        assert (
+            script_utxos2[0].amount == script_fund
+        ), f"Incorrect balance for script address `{script_utxos2[0].address}`"
+
+        # Step 2: spend the "locked" UTxOs
+
+        assert plutus_op1.datum_file and plutus_op2.datum_file
```

Comment:
missing assert error message like in the others above, also why this assertions are after the comment` # Step 2: spend the "locked" UTxOs` ?

Reply:
`datum_file` can be `None` according to the definition in corresponding named tuple, but it never can be `None` here, see lines 682 and 687. However `mypy` checker is not clever enough to see this. By adding the asserts, we assure `mypy` that the values are not `None`. It is better than `# type: ignore`, but it has similar purpose. The asserts are located here, because `mypy` use this assurance for lines 756 and 772.